KIPs/kip-81.md

```diff
@@ -0,0 +1,335 @@
+- **CnStakingV2**: An updated version of the [existing CnStakingContract](https://github.com/klaytn/klaytn/blob/v1.9.1/contracts/cnstaking/CnStakingContract.sol). V2 adds the balance notification and voter account appointing features.
+- **StakingTracker**: A new contract that tracks voting-related information from AddressBook and CnStakingV2 contracts. Accumulated information is served to Voting contract.
+- **Voting**: A new contract that oversees the on-chain voting process. Voting contract stores proposals, records votes, and sends transactions.
+
+Below digram shows relationships between contracts and accounts.
+
+![smart contract relation](../assets/kip-81/smart_contract_relation.png)
+
+From here, the contracts and accounts are explained.
+
+#### Voting contract
+
+The Voting contract is where voting happens. In the Voting contract, governance proposals are stored, casted votes are accumulated, and on-chain transactions are sent. The accounts interacting with the Voting contract are classified into proposer, voter, or executor roles. Its design is influenced by [Compound Finance](https://github.com/compound-finance/compound-protocol) and [OpenZeppelin](https://docs.openzeppelin.com/contracts/4.x/api/governance).
+
+The Voting contract revolves around proposals. A proposal describes the changes to be made on the Klaytn blockchain. A proposal not only contains textual descriptions of the change, but it may also include specifications of several on-chain transactions, or actions. The actions are executed on behalf of a voting contract after the proposal passes. The actions allow governance voting to enforce changes on the blockchain.
+
+```solidity
+abstract contract Voting {
+  struct Proposal {
+    string description;  // human readable description
+    address[] targets;   // 0 or more transactions
+    uint256[] values;
+    bytes[] calldatas;
+  }
+}
+```
+
+A proposer can submit governance proposals by calling `propose()` function. The voting timeline starts with the submission of proposals to the voting contract. A proposer can also cancel a proposal she has submitted, if the proposal had not been executed yet.
+
+A voter casts votes on behalf of one of the GCs. The vote choice is one of yes, no, or abstain.
+
+An executor triggers the execution of attached transactions in passed proposals. The execution is subdivided into two steps. First, the executor queues the transactions. After a set amount of delay, the executor can send the transactions. Execution delay here gives enough time for the community to recognize the upcoming change and perform a final check about the transaction.
+
+There will be a secretariat account stored in the Voting contract. The account is initially managed by the Klaytn Foundation, and later it can be replaced by governance voting. The secretariat has the proposer and executor role.
+
+```solidity
+abstract contract Voting {
```

Comment:
Can we specify more about how we can propose, queue, and execute the proposals?

Reply:
I'm not sure how the transactions actually executed on-chain with this info.

Reply:
I added a code snippet that executes the transactions from arrays.
Please review the "executing transactions" section in updated document (https://github.com/yeri-lee/kips/blob/master/KIPs/kip-81.md#executing-transactions)

Reply:
Using arrays of (target, value, calldata) is a common way of storing and executing multiple transactions.
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/Governor.sol#L316
https://github.com/makerdao/multicall/blob/master/src/Multicall.sol